machine-learning/03-Training and AutoML.ipynb

```diff
@@ -1,45 +1,561 @@
-{
-  "cells": [
-    {
-      "cell_type": "markdown",
-      "metadata": {},
-      "source": [
-        "# Training and AutoML"
-      ]
-    },
-    {
-      "cell_type": "markdown",
-      "metadata": {},
-      "source": [
-        "# Under Contstruction! Coming soon!"
-      ]
-    },
-    {
-      "cell_type": "markdown",
-      "metadata": {},
-      "source": [
-        "# Continue learning\n",
-        "\n",
-        "> [⏩ Next Module - Model Evaluation](https://raw.githubusercontent.com/JakeRadMSFT/csharp-notebooks/main/machine-learning/04-Model%20Evaluation.ipynb)  \n",
-        "> [⏪ Last Module - Data Prep and Feature Engineering](https://raw.githubusercontent.com/JakeRadMSFT/csharp-notebooks/main/machine-learning/02-Data%20Preparation%20and%20Feature%20Engineering.ipynb)  \n",
-        ""
-      ]
-    }
-  ],
-  "metadata": {
-    "kernelspec": {
-      "display_name": ".NET (C#)",
-      "language": "C#",
-      "name": ".net-csharp"
-    },
-    "language_info": {
-      "file_extension": ".cs",
-      "mimetype": "text/x-csharp",
-      "name": "C#",
-      "pygments_lexer": "csharp",
-      "version": "8.0"
-    }
-  },
-  "nbformat": 4,
-  "nbformat_minor": 4
-}
\ No newline at end of file
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "# Training and AutoML"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## In this Notebook, you will learn\n",
+        "- What is __Training__ actually mean in machine learning\n",
+        "- Available trainers in ML.Net, their difference, and how to decide which one to use.\n",
+        "- How hyper-parameter affects training performance.\n",
+        "- How to use built-in AutoML to simplify your training process.\n",
+        "\n",
+        "To be clarify, this notebook only talks about surpervised learning, so the training process will be performed on labeled data."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## What is __Training__ actually mean.\n",
+        "Before diving into code, let's first talk a little about what does \"train a model\" actually mean. \n",
+        "\n",
+        "From code aspect, \"train a model\" usually means call `model.fit(X, y)` in most of python machine learning packages, where `X` is feature array and `y` is label, or `model.Fit(X)` in ML.Net, where `X` is an `IDataView` which includes both feature and label.\n",
```

Comment:
```suggestion
        "## What is __Training__ actually mean.\n",
        "From code aspect, \"train a model\" usually means call `model.fit(X, y)` in most of python machine learning packages, where `X` is feature array and `y` is label, or `model.Fit(X)` in ML.Net, where `X` is an `IDataView` which includes both feature and label.\n",
```

Reply:
Is `Algorithm` the word we use in ML.Net to refer `trainer` or we should stick to use trainer instead? As `Algorithm` is a very vast concept